method `emit_fatal_errors` is private

Fresh rust, fresh cargo-src, doesn't work:

```
   cargo install cargo-src
   ...
   Compiling want v0.0.4
   Compiling rustdoc-highlight v0.1.7
   Compiling syn v0.11.11
error[E0624]: method `emit_fatal_errors` is private
   --> /home/rust/.cargo/registry/src/github.com-1ecc6299db9ec823/rustdoc-highlight-0.1.7/src/lib.rs:200:32
    |
200 |                     self.lexer.emit_fatal_errors();
    |                                ^^^^^^^^^^^^^^^^^

error[E0616]: field `sess` of struct `syntax::parse::lexer::StringReader` is private
   --> /home/rust/.cargo/registry/src/github.com-1ecc6299db9ec823/rustdoc-highlight-0.1.7/src/lib.rs:201:21
    |
201 |                     self.lexer.sess.span_diagnostic
    |                     ^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0616, E0624.
For more information about an error, try `rustc --explain E0616`.
error: Could not compile `rustdoc-highlight`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-src v0.1.6`, intermediate artifacts can be found at `/tmp/cargo-installvSQWyx`

Caused by:
  build failed
rust@desk:~$ rustc --version
rustc 1.36.0-nightly (a784a8022 2019-05-09)
rust@desk:~$ 

```

Would be nice if you provided the executable for Ubuntu x86_64 in Releases section, as compiling from sources or installing doesnt work.

--- a/src/highlight.rs
+++ b/src/highlight.rs
@@ -18,14 +18,21 @@ use rustdoc_highlight::{self as highlight, Class, Classifier};
 use span;
 use syntax;
 use syntax::parse;
-use syntax::parse::lexer::{self, TokenAndSpan};
+use syntax::parse::token::Token;
+use syntax::parse::lexer::{self};
 use syntax::source_map::{SourceMap, FilePathMapping, Loc};
 use syntax_pos::FileName;
+use syntax_pos::edition;
 
 use analysis::AnalysisHost;
 use analysis::DefKind;
 
+
 type Span = span::Span<span::ZeroIndexed>;
+
+pub fn with_globals<R>(f: impl FnOnce() -> R) -> R {
+    syntax::with_globals(edition::DEFAULT_EDITION, f)
+}
 
 pub fn highlight<'a>(
     analysis: &'a AnalysisHost,
@@ -35,7 +42,7 @@ pub fn highlight<'a>(
 ) -> String {
     debug!("highlight `{}` in `{}`", file_text, file_name);
 
-    syntax::with_globals(move || {
+    with_globals( || {
         let sess = parse::ParseSess::new(FilePathMapping::empty());
         let fm = sess.source_map().new_source_file(FileName::Real(PathBuf::from(&file_name)), file_text);
 
@@ -189,7 +196,7 @@ impl<'a> highlight::Writer for Highlighter<'a> {
         &mut self,
         text: T,
         klass: Class,
-        tas: Option<&TokenAndSpan>,
+        tas: Option<&Token>,
     ) -> io::Result<()> {
         let text = text.to_string();
 
@@ -198,8 +205,8 @@ impl<'a> highlight::Writer for Highlighter<'a> {
             Class::Ident | Class::Self_ => {
                 match tas {
                     Some(t) => {
-                        let lo = self.source_map.lookup_char_pos(t.sp.lo());
-                        let hi = self.source_map.lookup_char_pos(t.sp.hi());
+                        let lo = self.source_map.lookup_char_pos(t.span.lo());
+                        let hi = self.source_map.lookup_char_pos(t.span.hi());
                         // FIXME should be able to get all this info with a single query of analysis
                         let span = &self.span_from_locs(&lo, &hi);
                         let ty = self.analysis
@@ -276,8 +283,8 @@ impl<'a> highlight::Writer for Highlighter<'a> {
             }
             Class::RefKeyWord if text == "*" => match tas {
                 Some(t) => {
-                    let lo = self.source_map.lookup_char_pos(t.sp.lo());
-                    let hi = self.source_map.lookup_char_pos(t.sp.hi());
+                    let lo = self.source_map.lookup_char_pos(t.span.lo());
+                    let hi = self.source_map.lookup_char_pos(t.span.hi());
                     let span = &self.span_from_locs(&lo, &hi);
                     let mut extra = HashMap::new();
                     extra.insert(

--- a/src/server.rs
+++ b/src/server.rs
@@ -104,7 +104,7 @@ impl Service for Instance {
     type Request = Request;
     type Response = Response;
     type Error = Error;
-    type Future = Box<Future<Item = Self::Response, Error = Self::Error>>;
+    type Future = Box<dyn Future<Item = Self::Response, Error = Self::Error>>;
 
     fn call(&self, req: Request) -> Self::Future {
         let uri = req.uri().clone();

